### DISCUSSION TITLE
Streamlit and Nicegui viewport differance

### DISCUSSION THREAD

**Discussion by muratbedir** [zauberzeug/nicegui#1876] [Q&A]
### Description

I am moving stereamlit to nicegui. I stuck on viewport. In below screenshot left side Streamlit sample app. right side nicegui sample application. I want seen nicequi application like streamlit. 

On the steamlit a viewport bu I can not find how to apply same thing nicegui
`<meta name="viewport" content="width=device-width,initial-scale=1,shrink-to-fit=no">`

![image](https://github.com/zauberzeug/nicegui/assets/48950062/8b0cef70-4611-4aaf-b9ee-6170de40f225)

mine code below thanks for help. 

```py
#!/usr/bin/env python3
import nicegui.globals
from nicegui import app, ui

@ui.page('/')
def main_page() -> None:
    with ui.dialog() as about, ui.card():
        ui.label('About')
        ui.button('Close', on_click=about.close)
    menu(about)
    with ui.column().classes('absolute-center items-center') :        
        ui.label("SMS test uygulaması").classes('text-2xl text-positive')
        with ui.input("Telefon Numarası giriniz",  validation={'11 haneli telefon girini 0530112233 gibi': lambda value: (  False if len(str(value)) < 11 or len(str(value)) >11   else True ) },on_change=lambda e: badge.set_text(int(len(str(e.value))))).classes('text-xl').props('standout') as telefon:
            badge = ui.badge('0', color='red').props('floating outlined')
        sms=ui.textarea("SMS Metnini giriniz",placeholder="SMS test metnini giriniz",value="", validation={'Min 5 , Max 160 Char giriniz': lambda value: (len(value) < 160) and (len(value) > 5) }).classes('text-xl').props('standout')

        ui.label("SMS test uygulaması").classes('text-2xl text-positive')
        ui.button('open dialog',on_click=about.open).props('outline round')

@ui.page('/subpage')
def test_page() -> None:
    ui.label('This is a sub page.')

def menu(about):
    with ui.row().classes('w-full items-center'):
        result = ui.label().classes('mr-auto')
        with ui.button(icon='menu'):
            with ui.menu() as menu:
                ui.menu_item('About', on_click=about.open )
                ui.menu_item('Google 2', lambda:  ui.open("https://www.google.com") )
                ui.menu_item('Menu item 3', lambda: result.set_text('Selected item 3'))

ui.run(storage_secret='THIS_NEEDS_TO_BE_CHANGED',port=7654)`
```

**Comment by falkoschindler** [comment]
There is a `viewport` parameter in [`ui.run`](https://nicegui.io/documentation/run). You should be able to set it to "width=device-width,initial-scale=1,shrink-to-fit=no".

**Comment by muratbedir** [comment]
Thanks for help,

I add iu.run viewport parameter but still same, controls not stretched.

is problem can be  related to  below code ? 
`with ui.column().classes('absolute-center items-center') :   ` 

my iu.run viewport parameter after changed

`ui.run(storage_secret='THIS_NEEDS_TO_BE_CHANGED',port=7654,viewport='width=device-width,initial-scale=1,shrink-to-fit=no')`

screen shot (full screen) in below.  I want to stretch controls at least beyond the red line (I want because its look better) 

![image](https://github.com/zauberzeug/nicegui/assets/48950062/7812e6f8-1b22-40e3-a470-50fbe426f175)

**Reply by falkoschindler** [reply]
I don't understand what you mean with the red lines. But if you want to stretch the input fields horizontally, you can set a column width an stretch its content like this:
```py
with ui.column().classes('absolute-center items-stretch w-[400px]'):
    ui.label("SMS test uygulaması").classes('text-2xl text-positive')
    with ui.input("Telefon Numarası giriniz").classes('text-xl').props('standout'):
        ui.badge('0', color='red').props('floating outlined')
    ui.textarea("SMS Metnini giriniz", placeholder="SMS test metnini giriniz").classes('text-xl').props('standout')

    ui.label("SMS test uygulaması").classes('text-2xl text-positive')
    ui.button('open dialog').props('outline rounded')
```

**Reply by natankeddem** [reply]
Is there an advantage to manipulating viewport rather than just playing around with ui containers?
